### PR TITLE
RAB-686: Handle error and move StorageClientInterface into Domain

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Infrastructure/EventSubscriber/UpdateJobExecutionStorageSummarySubscriber.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     * @return array<string, string>
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            FileCannotBeExported::class => 'onfileCannotBeExported',
+        ];
+    }
+
+    public function onfileCannotBeExported(FileCannotBeExported $event): void
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Infrastructure/EventSubscriber/UpdateJobExecutionStorageSummarySubscriber.php:33]:
```suggestion
    public function onFileCannotBeExported(FileCannotBeExported $event): void
```

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Domain/Event/FileCannotBeExported.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * (c) 2022 Akeneo SAS (https://www.akeneo.com)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Akeneo\Platform\Bundle\ImportExportBundle\Domain\Event;
+
+class FileCannotBeExported
```
Comment (grena) [inline src/Akeneo/Platform/Bundle/ImportExportBundle/Domain/Event/FileCannotBeExported.php:16]:
```suggestion
class FileTransferFailed
```
It's just a suggestion.
`FileCannotBeExported` looks like it misses some configuration or data to launch the export

**Review by grena — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/ImportExportBundle/Domain/Event/FileCannotBeExported.php (hunk 2 above)
Comment (ghoussard) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/Domain/Event/FileCannotBeExported.php:16]: FileTransferFailed might be confusing later if we have an another event for file tranfer before import. I let it like that in this PR. 👍🏻

Comment (grena) [reply src/Akeneo/Platform/Bundle/ImportExportBundle/Domain/Event/FileCannotBeExported.php:16]: I see, it makes sense then!

**Pull request merged**